### PR TITLE
Enable includes

### PR DESCRIPTION
**Pull request by zrpnr**
This fixes the attributes enabled for top level, and enabled attributes for included resources.
Added a checkbox to "include" a relationship, this works as a toggle so they can be added or removed.
Multiple relationships can be included, as well as deeply nested relationships.

Fields are fixed as well so that fields from different resources are toggled separately.

example url: ```http://drupal.test/jsonapi/node/article?include=node_type,uid.roles&fields[node--article]=drupal_internal__nid&fields[node_type--node_type]=drupal_internal__type,name&fields[user_role--user_role]=drupal_internal__id```

### CODE REVIEW

File: src/schema.js [hunk 1]
```diff
@@ -9,14 +9,17 @@ import {
...
 
 const Schema = ({ url, includePath = [] }) => {
   const [type, setType] = useState('');
   const [attributes, setAttributes] = useState([]);
   const [relationships, setRelationships] = useState([]);
-  const { include } = useContext(LocationContext);
+  const { include, toggleInclude } = useContext(LocationContext);
+
+  const includesEnabled = checkIncludesPath(include, includePath);
+  const includePathDisplay = includePath.join('.');
```
Comment (gabesullice) [inline src/schema.js:22]: Maybe this should be `includePathString` because it's not only for display, it's only used for `onChange={() => toggleInclude(includePathDisplay)}`

File: src/utils.js [hunk 2]
```diff
@@ -13,4 +13,12 @@ export function toggleSetEntry(set, entry) {
   } else {
     set.add(entry);
   }
+
+  return set;
```
Comment (gabesullice) [inline src/utils.js:17]: Nice, this changes this method into a more fluent style 👍

File: src/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  test('Relationship: mismatch include', () => {
+    expect(checkIncludesPath(['node_type'], ['uid'])).toBe(false);
+  });
+
+  test('Relationship: matching include plus other', () => {
+    expect(checkIncludesPath(['uid', 'node_type'], ['uid'])).toBe(true);
+  });
+
+  test('Relationship: matching deep include', () => {
+    expect(checkIncludesPath(['uid.roles'], ['uid', 'roles'])).toBe(true);
```
Comment (gabesullice) [inline src/utils.test.js:33]: Awesome! I came here expecting to make this suggestion because it's kind of an edge case in the spec, but you already saw it 👍

**Review by gabesullice — APPROVED** [review]
Awesome. I like this. Merging.

File: src/schema.js (hunk 1 above)
Comment (zrpnr) [reply src/schema.js:22]: can do, I'll just commit and push this.
